feat: add Discord integration tool

vishkrish200 opened the pull request:
Add Discord bot functionality to phi tools with async support. Includes basic channel and message management capabilities.

- Send/delete messages
- Get channel history
- List channels and info
- Example usage in cookbook

this fixes #1330

vishkrish200 commented:
had some issues with the validate script. had to deal with the async nature of the discord.py library

anuragts commented on phi/tools/discord_tools.py:
```diff
@@ -0,0 +1,195 @@
+        """
+        super().__init__(name="discord_tools")
+
+        self.bot_token = bot_token
```
Can you also add get the bot_token from env. 
example - getenv("EXA_API_KEY")
Also add logger.error if bot token is not found.

vishkrish200 replied:
hey solved for this in the latest commit

anuragts commented on cookbook/tools/discord_tools.py:
```diff
@@ -0,0 +1,43 @@
+
+# Create an agent with the discord toolkit
+agent = Agent(
+    instructions=[
```
Add a unique name and pass model to the agent, also set markdown to true.
here is an example - https://github.com/phidatahq/phidata/blob/main/cookbook/agents_101/web_search.py

vishkrish200 replied:
solved for this too

vishkrish200 commented:
@anuragts . just did some more testing of the tool and it looks like it will need a few more changes to work. 

discord.py is inherently asynchronous - it requires a running event loop.
The discord bot needs to be properly started and connected before any operations.
Apologies for the delay

vishkrish200 commented:
hey the latest commits fix everything. switched the approach completely. using discord's http api instead of the discord bot system. updated cookbook with the appropriate steps. also included the changes @anuragts requested :)

vishkrish200 commented:
hey @anuragts, the pr says changes are requested but those changes were marked as resolved by you. could you have a look at the updated code again?

manthanguptaa commented:
Thanks @vishkrish200 for this amazing PR. I have tested it out and it's working perfectly!

manthanguptaa commented on cookbook/tools/discord_tools.py:
```diff
@@ -0,0 +1,54 @@
+"""Example showing how to use Discord tools."""
```
you can remove this

vishkrish200 reviewed (commented):
resolved the issue @manthanguptaa. removed the comment 👍

manthanguptaa commented:
@vishkrish200 can you also edit the doc strings for the function to add the function definition. Example https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py

vishkrish200 commented:
> @vishkrish200 can you also edit the doc strings for the function to add the function definition. Example https://github.com/phidatahq/phidata/blob/main/phi/tools/slack.py

hey just pushed a commit which fixes this. Let me know if there are any other changes needed. also i think i've unnecessarily requested a pr review from @anuragts.

vishkrish200 commented:
@manthanguptaa any updates?
